### ISSUE TITLE
Use txt extension for text documents

### ISSUE THREAD

**Issue by vkbo** [vkbo/novelWriter#1964]
When implementing #977, also rename the content files to `.txt`. The current extension, `.nwd` was mostly there to obscure the format, but it doesn't really do that (see #1960) and `.txt` is technically more correct. The build format is also `.txt` now.

**Comment by HeyMyian** [comment]
Switching from `.nwd` to `.txt` will break the git diff continuum, won't it? For this reason I would prefer a user-controlled setting that converts `.nwd` to `.txt` only if the user wants to, or maybe in combination with auto-detect as to which extension the files currently use. New projects and/or freshly installed novelWriters may default to `.txt`.

**Comment by vkbo** [comment]
> Switching from `.nwd` to `.txt` will break the git diff continuum, won't it?

No, git is perfectly capable of handling file renames as long as the content is the same.

> For this reason I would prefer a user-controlled setting that converts `.nwd` to `.txt` only if the user wants to, or maybe in combination with auto-detect as to which extension the files currently use.

How the project is structured changes from time to time. This is a continuous process depending on in-app changes. The user shouldn't be concerned with the project file structure at all. Version control is designed to handle changes, so that's a non-issue.

> New projects and/or freshly installed novelWriters may default to `.txt`.

It's either all or nothing. Maintaining multiple structures is asking for trouble.

**Comment by vkbo** [comment]
I've decided to implement this as a part of #977, and do a major version bump on the file format to 2.0. It's a good opportunity to redesign the project format and convert the project file to JSON and rename the content files to `.txt` and let it all be handled by a one-time, one-way conversion.

**Comment by awqk** [comment]
I'm wondering if `.md` might be better than `.txt`. It's often associated with markdown text.

This would allow to set up a text converter in git config for `.md` files, for example to insert newlines between sentences to improve overall `git diff` output.

**Comment by vkbo** [comment]
They are not Markdown files.

**Comment by vkbo** [comment]
I'm picking this up again as part of a rewrite of the document class. I'm moving away from custom formats and adapting more standard document formats. For the novelWriter documents it means renaming them to `.md` as @awqk suggests, and adding the meta data as a standard toml header. I'm avoiding the more common yaml header to avoid the new dependency since Python since 3.11 has a built-in toml parser.

All documents will be converted on start, so this means a one way update when it hits live users.
